keyless/circuit-data/templates/helpers/jwt_field_parsing.circom

```diff
@@ -0,0 +1,131 @@
+pragma circom 2.1.3;
+
+include "helpers/misc.circom";
+include "helpers/arrays.circom";
+include "helpers/hashtofield.circom";
+include "helpers/packing.circom";
+include "../node_modules/circomlib/circuits/gates.circom";
+include "../node_modules/circomlib/circuits/bitify.circom";
+
+// Checks the given jwt key value pair has a colon in between the name and value, a comma or endbrace at the end, and only whitespace in between the name and colon, colon and value, and value and end character. Returns the name and value fields 
+// We did this instead of a polynomial concatenation check to avoid having to implement a multi-variable concatenation check
+template ParseJWTFieldWithQuotedValue(maxKVPairLen, maxNameLen, maxValueLen) {
+    signal input field[maxKVPairLen]; // ASCII
+    signal input name[maxNameLen];
+    signal input value[maxValueLen];
+    signal input field_len; // ASCII
+    signal input name_len;
+    signal input value_index; // index of value within `field`
+    signal input value_len;
+    signal input colon_index; // index of colon within `field`
+
+    // Enforce that end of name < colon < start of value and that field_len >=
+    // name_len + value_len + 1 (where the +1 is for the colon), so that the
+    // parts of the JWT field are in the correct order
+    signal colon_greater_name <== LessThan(20)([name_len, colon_index]);
+    colon_greater_name === 1;
+    signal colon_less_value <== LessThan(20)([colon_index, value_index]);
+    colon_less_value === 1;
+    signal field_len_ok <== GreaterEqThan(20)([field_len, name_len + value_len + 1]);
+    field_len_ok === 1;
+
+    signal field_hash <== HashBytesToFieldWithLen(maxKVPairLen)(field, field_len);
+
+    signal name_first_quote <== SelectArrayValue(maxKVPairLen)(field, 0);
+    name_first_quote === 34; // '"'
+    CheckSubstrInclusionPoly(maxKVPairLen, maxNameLen)(field, field_hash, name, name_len, 1);
+    signal name_second_quote <== SelectArrayValue(maxKVPairLen)(field, name_len+1);
+    name_second_quote === 34; // '"'
+
+    signal colon <== SelectArrayValue(maxKVPairLen)(field, colon_index);
+    colon === 58; // ':'
+
+    signal value_first_quote <== SelectArrayValue(maxKVPairLen)(field, value_index-1);
```

Comment:
This would be cleaner if we called `ParseJWTFieldWithUnquotedValue` here and then added the quote value checking logic on top of it